<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UserEditor</class>
 <widget class="QWidget" name="UserEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>601</width>
    <height>636</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QDockWidget" name="simPane">
     <property name="floating">
      <bool>false</bool>
     </property>
     <property name="windowTitle">
      <string>Simulation results</string>
     </property>
     <widget class="QWidget" name="dockWidgetContents_2">
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="frame_4">
         <property name="frameShape">
          <enum>QFrame::HLine</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QGroupBox" name="groupBox_5">
           <property name="title">
            <string>Timing</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <property name="verticalSpacing">
               <number>0</number>
              </property>
              <item row="0" column="1">
               <widget class="QLabel" name="simTotalTime">
                <property name="font">
                 <font>
                  <weight>75</weight>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string>0 s</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Minimum runtime:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Finished:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLabel" name="simFinished">
                <property name="font">
                 <font>
                  <weight>75</weight>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string>0 s</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Ignored</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QGroupBox" name="groupBox_6">
           <property name="title">
            <string>Device ranges</string>
           </property>
           <property name="checkable">
            <bool>false</bool>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_9">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QTreeWidget" name="simRanges">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>100</height>
               </size>
              </property>
              <property name="rootIsDecorated">
               <bool>false</bool>
              </property>
              <property name="allColumnsShowFocus">
               <bool>true</bool>
              </property>
              <property name="headerHidden">
               <bool>true</bool>
              </property>
              <property name="columnCount">
               <number>4</number>
              </property>
              <attribute name="headerDefaultSectionSize">
               <number>50</number>
              </attribute>
              <attribute name="headerMinimumSectionSize">
               <number>20</number>
              </attribute>
              <column>
               <property name="text">
                <string notr="true">device</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>min</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string/>
               </property>
              </column>
              <column>
               <property name="text">
                <string>max</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Output</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCheckBox" name="simErrorsOnly">
             <property name="text">
              <string>Errors/warnings only</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QStackedWidget" name="simOutStack">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="simOutNormalPage">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="OutputView" name="simOutView">
               <property name="autoFillBackground">
                <bool>true</bool>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="tabChangesFocus">
                <bool>true</bool>
               </property>
               <property name="lineWrapMode">
                <enum>QTextEdit::NoWrap</enum>
               </property>
               <property name="openLinks">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="simOutErrorPage">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="OutputView" name="simOutViewErrors">
               <property name="autoFillBackground">
                <bool>true</bool>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="tabChangesFocus">
                <bool>true</bool>
               </property>
               <property name="lineWrapMode">
                <enum>QTextEdit::NoWrap</enum>
               </property>
               <property name="openLinks">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QDockWidget" name="scriptsPane">
      <property name="features">
       <set>QDockWidget::DockWidgetClosable|QDockWidget::DockWidgetMovable</set>
      </property>
      <property name="allowedAreas">
       <set>Qt::LeftDockWidgetArea</set>
      </property>
      <property name="windowTitle">
       <string>Scripts</string>
      </property>
      <widget class="QWidget" name="dockWidgetContents">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QTreeView" name="fileTree"/>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QFrame" name="mainFrame">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDockWidget" name="warnWidget">
     <property name="palette">
      <palette>
       <active>
        <colorrole role="Button">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>127</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Base">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>255</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Window">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>127</blue>
          </color>
         </brush>
        </colorrole>
       </active>
       <inactive>
        <colorrole role="Button">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>127</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Base">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>255</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Window">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>127</blue>
          </color>
         </brush>
        </colorrole>
       </inactive>
       <disabled>
        <colorrole role="Button">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>127</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Base">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>127</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Window">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>255</red>
           <green>255</green>
           <blue>127</blue>
          </color>
         </brush>
        </colorrole>
       </disabled>
      </palette>
     </property>
     <property name="floating">
      <bool>false</bool>
     </property>
     <property name="features">
      <set>QDockWidget::DockWidgetClosable|QDockWidget::DockWidgetMovable</set>
     </property>
     <property name="windowTitle">
      <string>Warning</string>
     </property>
     <widget class="QWidget" name="dockWidgetContents_3">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="warnText">
         <property name="text">
          <string>...</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/open</normaloff>:/open</iconset>
   </property>
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="toolTip">
    <string>Open script from disk</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/new</normaloff>:/new</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="toolTip">
    <string>New script</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/save</normaloff>:/save</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="toolTip">
    <string>Save script to disk</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save &amp;as...</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/print</normaloff>:/print</iconset>
   </property>
   <property name="text">
    <string>&amp;Print...</string>
   </property>
   <property name="toolTip">
    <string>Print script</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/undo</normaloff>:/undo</iconset>
   </property>
   <property name="text">
    <string>&amp;Undo</string>
   </property>
   <property name="toolTip">
    <string>Undo edit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/redo</normaloff>:/redo</iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
   <property name="toolTip">
    <string>Redo edit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/cut</normaloff>:/cut</iconset>
   </property>
   <property name="text">
    <string>&amp;Cut</string>
   </property>
   <property name="toolTip">
    <string>Cut selected text to clipboard</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/copy</normaloff>:/copy</iconset>
   </property>
   <property name="text">
    <string>C&amp;opy</string>
   </property>
   <property name="toolTip">
    <string>Copy selected text to clipboard</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/paste</normaloff>:/paste</iconset>
   </property>
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="toolTip">
    <string>Paste text from clipboard</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/continue</normaloff>:/continue</iconset>
   </property>
   <property name="text">
    <string>&amp;Run</string>
   </property>
   <property name="toolTip">
    <string>Run the script in this window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Return</string>
   </property>
  </action>
  <action name="actionGet">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/get</normaloff>:/get</iconset>
   </property>
   <property name="text">
    <string>&amp;Get</string>
   </property>
   <property name="toolTip">
    <string>Copy the currently running script into this window</string>
   </property>
  </action>
  <action name="actionFind">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/find</normaloff>:/find</iconset>
   </property>
   <property name="text">
    <string>&amp;Find...</string>
   </property>
   <property name="toolTip">
    <string>Find and replace text</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionComment">
   <property name="text">
    <string>Co&amp;mment</string>
   </property>
   <property name="toolTip">
    <string>Comment out selected code</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+#</string>
   </property>
  </action>
  <action name="actionUncomment">
   <property name="text">
    <string>Uncomment</string>
   </property>
   <property name="toolTip">
    <string>Uncomment selected code</string>
   </property>
   <property name="shortcut">
    <string>Alt+#</string>
   </property>
  </action>
  <action name="actionUpdate">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/refresh</normaloff>:/refresh</iconset>
   </property>
   <property name="text">
    <string>&amp;Update</string>
   </property>
   <property name="toolTip">
    <string>Update the currently running script with the one in this window</string>
   </property>
  </action>
  <action name="actionReload">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Re&amp;load from disk</string>
   </property>
  </action>
  <action name="actionSimulate">
   <property name="icon">
    <iconset resource="../../../../resources/nicos-gui.qrc">
     <normaloff>:/simulate</normaloff>:/simulate</iconset>
   </property>
   <property name="text">
    <string>Simulate</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Return</string>
   </property>
  </action>
  <action name="actionShowScripts">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show scripts</string>
   </property>
   <property name="toolTip">
    <string>Show scripts browser</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OutputView</class>
   <extends>QTextBrowser</extends>
   <header>nicos.gui.outview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../resources/nicos-gui.qrc"/>
 </resources>
 <connections/>
</ui>
